src/core/qgsaggregatecalculator.h

```diff
@@ -136,6 +137,22 @@ class CORE_EXPORT QgsAggregateCalculator
      */
     void setFilter( const QString &filterExpression ) { mFilterExpression = filterExpression; }
 
+    /**
+     * Sets a filter to limit the features used during the aggregate calculation.
+     * If an expression filter is set, it will override this filter unless filter stacking is enabled (sackFilters)
+     * \param  fids feature ids for feature filtering, and empty list will return no features.
+     * \see filter()
+     */
+    void setFidsFilter( const QgsFeatureIds fids );
+
+    /**
+     * Endable or disable filter stacking, if disabled the expression filter will override the Fids filter if both are set.
+     * When enables the inner set of the expression and fids will be used.
+     * \param stack boolean to enable or disable filter stacking.
+     * \see filter()
+     */
+    void stackFilters( bool stack ) { mStackFilters = stack; }
```

Comment:
Can you remove this until after #9915 is finalised please?

Reply:
Done, thanks again for the review!